### PR TITLE
Feature/e UI 4606 date year validation

### PR DESCRIPTION
**Pull request by phillip-whitaker-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-4606

### Change description ###

Enhanced the date year input validation to restrict between 1900-2100

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by Daniel-Lam — COMMENTED** [review]
@phillip-whitaker-hmcts - changes look OK. Minor suggestion: how about adding some boundary tests, i.e. when the date is either `1900-01-01T00:00:00.000` or `2099-12-31T23:59:59.999`?

**Review by RiteshHMCTS — DISMISSED** [review]

File: src/components/form/date-input/date-input-component.spec.ts [hunk 1]
```diff
@@ -64,13 +64,18 @@ describe('Date input component', function () {
...
+    expect(results).toBeUndefined();
+  });
+
   it('should be invalid when the date is before 1900', () => {
-    const results = component.validate({ value: '1891-04-09T08:02:27.542' } as FormControl);
+    const results = component.validate({ value: '1891-12-31T12:59:59.999' } as FormControl);
     expect(results).toEqual({ pattern: 'Date is not valid' });
   });
 
   it('should be invalid when the date is after 2100', () => {
```
[the remark this answers is not in the document]
Comment (phillip-whitaker-hmcts) [reply src/components/form/date-input/date-input-component.spec.ts]: @Daniel-Lam I have changed the test wording :)

[the remark this answers is not in the document]
Comment (Daniel-Lam) [reply src/components/form/date-input/date-input-component.spec.ts]: @phillip-whitaker-hmcts - many thanks.

**Review by Daniel-Lam — DISMISSED** [review]

**Review by RiteshHMCTS — DISMISSED** [review]
Approved

**Review by RiteshHMCTS — APPROVED** [review]

**Review by Mo-Lala84 — APPROVED** [review]
approved

**Pull request merged**